bread/menu.py

```diff
@@ -100,6 +101,18 @@ def from_objectaction(object, actionname, label, icon=None, *args, **kwargs):
         )
 
 
+class Delete(Action):
+    def __init__(self):
+        from . import layout
+
+        super().__init__(
+            js=hg.F(
+                lambda c, e: f'window.location = \'{layout.objectaction(c["row"], "delete")}?next=\' + window.location.pathname + window.location.search',
+            ),
+            icon="trash-can",
+        )
+
+
```

Comment:
Can you explain the purpose of this class? I do not see it used in this PR?
I wonder if this is the right place to add this shortcut. I definitely see the need for this kind of shortcut to make the API simpler. But the problem with these kind of "object-action" is that they depend at least on the context variable of the according object ("row" in this case) and the availability of an object-action (or URL in the end), here "delete". I think the ```hg.F(....)``` should be subclassed in a generic way (that means it would be a ```htmlgenerator.Lazy``` object which can handle object-actions).
In any case I would keep the ```menu``` module free of dependencies to the ```layout``` module for now. The menu- and action-system anyway needs to be redesigned. One point is that the menu itself could easier be implemented by having the corresponding carbon design menu and then just adding that element to the template/render context. Another point is that Action and Link need possibly to be implemented inside the ```layout``` module. They both represent the abstract idea of an action a user can do in the UI. Decoupling that idea completely from the UI-code (the ```layout``` module) is likely not a very good idea.
But in any case, this would be a bit a larger refactoring, I just braindumped a bit here so you also get aware of the issue.

Reply:
I don't even understand this code fully :D
I just needed a "delete"-Button for each row and found something similar somewhere else in the code so I thought I am going to share the code in order to make explicit that I am doing the exact same thing (without understanding it).
I guess what I will do is to remove the Delete class and duplicate the F function. Are you ok with that?